packages/directory/addon/components/dir-new-button.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2018, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * {{dir-new-button}}
+ */
+
+import Component from '@ember/component';
+import layout from '../templates/components/dir-new-button';
+import { set } from '@ember/object';
+
+export default Component.extend({
+  layout,
+
+  classNames: [ 'dir-new-button' ],
+
+  /**
+   * @method calculatePosition
+   * @returns {Object} - positioning info used by ember-basic-dropdown
+   */
+  calculatePosition(trigger, content) {
+    let { top, left, width, height } = trigger.getBoundingClientRect(),
+        { width: contentWidth } = content.getBoundingClientRect(),
+        marginFromTopBar = 15,
+        style = {
+          left: left - contentWidth + width,
+          top: top + height + marginFromTopBar
+        };
+
+    return { style };
+  },
+
+  /**
+   * @override
+   * @method init
+   */
+  init() {
+    this._super(...arguments);
+
+    set(this, 'documentTypes', [ {name: 'reports', iconClass: 'file'}, {name: 'dashboards', iconClass: 'table'} ]);
```

Comment:
Why this cannot be an attr?

Reply:
If we make this an attribute, this same line of code will just have to be moved to the route that instantiates this component.

Reply:
Sorry, I mean a property of the component

Reply:
This is to avoid leaking state into the factory of this component. 
More info here: https://dockyard.com/blog/2015/09/18/ember-best-practices-avoid-leaking-state-into-factories

Balaji brought this up a couple weeks ago in the navi-dev slack channel

Reply:
I guess use `compute` can also resolve this issue and it is a little bit concise

Reply:
I don't feel like it makes much sense to have a computed property return a static array

Reply:
I guess this can be configurable value, so we may want to get documentTypes from enviroment.json

Reply:
The idea was that you could just override this method to change the list of `documentTypes`